test/Microsoft.AspNetCore.Dispatcher.Test/Tree/TreeMatcherTest.cs

```diff
@@ -776,6 +780,32 @@ public void MatchAsync_DoesNotMatchShorterUrl()
             Assert.Null(context.Endpoint);
         }
 
+        [Fact]
+        public async Task ChangeTokenTriggers_CreateCache()
+        {
+            // Arrange 1
+            var factory = new TreeMatcherFactory();
+            var dataSource = new TestDataSource(new List<Endpoint>() { new RoutePatternEndpoint("test/{p1}", new object(), Test_Delegate, "Test"), });
+            var matcher = factory.CreateMatcher(dataSource, new List<EndpointSelector>());
+            var context = CreateMatcherContext("/test/parameter1");
+
+            // Act 1
+            await matcher.MatchAsync(context);
+
+            // Assert 1
+            Assert.Equal("Test", context.Endpoint.DisplayName);
+
+            // Arrange 2
+            dataSource.UpdateEndpoints(new List<Endpoint>() { new RoutePatternEndpoint("test2/{p1}", new object(), Test_Delegate, "Test2"), });
+            context = CreateMatcherContext("/test2/parameter2");
+
+            // Act 2
+            await matcher.MatchAsync(context);
+
+            // Assert 2
+            Assert.Equal("Test2", context.Endpoint.DisplayName);
```

Comment:
I wanted to check if context.Values are populated with the key "p1" with value "parameter2" but context.Values is an empty collection. Is that expected?

Reply:
No, that doesn't seem right.